libs/scalalib/src/mill/scalalib/SonatypeCentralPublisher.scala

```diff
@@ -42,7 +42,26 @@ class SonatypeCentralPublisher(
   ): Unit = {
     val mappings = getArtifactMappings(isSigned = true, gpgArgs, workspace, env, artifacts)
     log.info(s"mappings ${pprint.apply(mappings.map { case (a, kvs) => (a, kvs.map(_._1)) })}")
-    val (_, releases) = mappings.partition(_._1.isSnapshot)
+    val (snapshots, releases) = mappings.partition(_._1.isSnapshot)
+    if (snapshots.nonEmpty) {
+      val snapshotNames = snapshots.map(_._1)
+        .map { case Artifact(group, id, version) => s"$group:$id:$version" }
+      log.warn(
+        s"""Publishing snapshots to Sonatype Central Portal is currently not supported by Mill.
+           |This is tracked under https://github.com/com-lihaoyi/mill/issues/4421
+           |The following snapshots will not be published:
+           |  ${snapshotNames.mkString("\n  ")}""".stripMargin
+      )
```

Comment:
Let's fail in this case, too. `Task.fail` is the convenient API to use here.

Reply:
done